packages/react-dom/src/events/__tests__/SyntheticEvent-test.js

```diff
@@ -18,63 +17,122 @@ describe('SyntheticEvent', () => {
   var createEvent;
 
   beforeEach(() => {
-    // TODO: can we express this test with only public API?
-    SyntheticEvent = require('SyntheticEvent');
     React = require('react');
     ReactDOM = require('react-dom');
     ReactTestUtils = require('react-dom/test-utils');
-
-    createEvent = function(nativeEvent) {
-      var target = require('getEventTarget')(nativeEvent);
-      return SyntheticEvent.getPooled({}, '', nativeEvent, target);
+    createEvent = (eventInit, eventOptions) => {
+      const defaultEventInit = {
+        bubbles: true,
+        cancelable: true,
+        ...eventOptions,
+      };
+      var event = new Event(eventInit, defaultEventInit);
+      event.initEvent(
+        eventInit,
+        defaultEventInit.bubbles,
+        defaultEventInit.cancelable,
+      );
+      return event;
     };
   });
 
   it('should normalize `target` from the nativeEvent', () => {
-    var target = document.createElement('div');
-    var syntheticEvent = createEvent({srcElement: target});
+    var click = jest.fn();
+    var container = document.createElement('div');
+
+    var onClick = e => click(e.target);
+
+    var instance = ReactDOM.render(<div onClick={onClick} />, container);
 
-    expect(syntheticEvent.target).toBe(target);
-    expect(syntheticEvent.type).toBe(undefined);
+    document.body.appendChild(container);
+
+    var event = createEvent('click');
+
+    instance.dispatchEvent(event);
+    expect(click).toBeCalledWith(instance);
+    document.body.removeChild(container);
   });
 
   it('should be able to `preventDefault`', () => {
-    var nativeEvent = {};
-    var syntheticEvent = createEvent(nativeEvent);
+    var click = jest.fn();
+    var container = document.createElement('div');
+
+    var onClick = e => {
+      click(e.isDefaultPrevented());
+      e.preventDefault();
+      click(e.isDefaultPrevented());
+      click(e.defaultPrevented);
+      click(e.nativeEvent.returnValue);
+    };
 
-    expect(syntheticEvent.isDefaultPrevented()).toBe(false);
-    syntheticEvent.preventDefault();
-    expect(syntheticEvent.isDefaultPrevented()).toBe(true);
+    var instance = ReactDOM.render(<div onClick={onClick} />, container);
 
-    expect(syntheticEvent.defaultPrevented).toBe(true);
+    document.body.appendChild(container);
 
-    expect(nativeEvent.returnValue).toBe(false);
+    var event = createEvent('click');
+    instance.dispatchEvent(event);
+    expect(click.mock.calls[0][0]).toBe(false);
```

Comment:
This is not testing anything in React. You are creating a jsdom event (equivalent to a browser event) and then check that jsdom was dispatched. Even if the entire React source code was deleted, this test would still pass.